### DISCUSSION TITLE
engine0.6.x版本如何实现小程序监听屏幕x轴触摸滑动事件，实现旋转模型

### DISCUSSION THREAD

**Discussion by KingCyril** [galacean/discussion#13] [Mini Program（小程序）] [answered]
由于hdr加载卡顿问题，淘宝小程序中oasis版本需退回0.6版本，下边inputManager不能使用了，如何在6版本实现这段代码

onUpdate(deltaTime: number): void {
    const rotateInstance = <any>this;
    const input = rotateInstance.engine.inputManager;
    /**
     * 操作中不允许旋转
     */
    if (rotateInstance.entity['_instance'].isOperating) return;
    if (
      input.isPointerHeldDown(PointerButton.Primary) &&
      input.pointerMovingDelta.x !== 0
    ) {
      const rot = isAndroid ? input.pointerMovingDelta.x : input.pointerMovingDelta.x / 5;
      rotateInstance.entity.transform.rotate(
        this.rotate.set(0, 0, rot)
      );
    }
  }

**Comment by KingCyril** [comment]
/**
   * 代理事件
   */
  const eventProps = {
    onTouchStart(e) {
      dispatchPointerDown(e);
    },
    onTouchMove(e) {
      dispatchPointerMove(e);
    },
    onTouchEnd(e) {
      dispatchPointerUp(e);
    },
  };

**Comment by cptbtptpbcptdtptp** [comment, marked as answer]
0.6 版本没有 pointerMovingDelta 这个属性，只能自己监听 touch 事件来实现了。

**Reply by cptbtptpbcptdtptp** [reply]
```
const rot = isAndroid ? input.pointerMovingDelta.x : input.pointerMovingDelta.x / 5;
```
这行代码的逻辑很奇怪，像是为了解决安卓和 IOS 体感上滑动相同距离但是旋转角度不同的问题，实际上不应该这么做，这个问题出现的本质原因是不同手机画布宽高不同导致的，pointerMovingDelta 的大小也是基于画布宽高的（比如画布宽 1000 ，你在移动了半屏，返回的X差值就是 500，同理画布宽 2000 返回的差值就是 1000），这里应该使用归一化来处理：
```
const rot = input.pointerMovingDelta.x / canvas.width * 180;
```

**Reply by KingCyril** [reply]
哦哦，之前不清楚pointerMovingDelta属性，所以就自己判断机型处理掉了，现在切换0.6后已经通过监听事件实现模型旋转了，谢谢了
